File: challenges/2/2.js
function fibonacci_evens(limit) {
    var fs = [1, 2];

    // create array of all fibonacci numbers below limit param
    while (fs[fs.length - 1] + fs[fs.length - 2] < limit) {
        fs.push(fs[fs.length - 1] + fs[fs.length - 2]);
    }

    // filter out odd numbers, then sum all array elements
    return fs.filter(idx => idx % 2 == 0).reduce((a, b) => a + b, 0);
}

console.log(fibonacci_evens(4000000));

File: challenges/2/2.py
def fibonacci_evens(limit):
    fs = [1, 2]

    # create array of all fibonacci numbers below limit param
    while fs[-1] + fs[-2] < limit:
        fs.append(fs[-1] + fs[-2])

    # sum all even numbers in array
    return(sum(n for n in fs if n % 2 == 0))


print(fibonacci_evens(4000000))
